libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -2037,6 +2057,38 @@ bool AP_Mission::get_next_do_cmd(uint16_t start_index, Mission_Command& cmd)
 /// jump handling methods
 ///
 
+// find the first JUMP_TAG with this tag and return its index.
+// Returns 0 if no appropriate JUMP_TAG match can be found.
+uint16_t AP_Mission::get_index_of_jump_tag(const uint16_t tag)
+{
+    for (uint16_t i = 1; i < (unsigned)_cmd_total; i++) {
```

Comment:
I don't think we can afford this loop at runtime.  Could you try uploading a *big* mission and getting some microsecond measurements for this lookup?

Big mission meaning fills storage and has the looked-for tag right at the end.

If we can't then having a table of jump tags -> cmd_id would be relatively straight-forward, I think

Reply:
Caching the tags is an option, but then it limits us.

Reply:
We do similar things for the DO_LAND_START. I agree that it should at least be measured. I'll do that. Meanwhile, I'm open to suggestions to improve it.

Reply:
> Big mission meaning fills storage and has the looked-for tag right at the end.

Storage is backed into RAM though, so while it would be bad to do often, the one off cost is probably worth it for the feature, but having hard timing data would be good.